metacat-main/src/main/java/com/netflix/metacat/main/configs/ServicesConfig.java

```diff
@@ -73,7 +74,9 @@ public UserMetadataService userMetadataService(
         final Config config,
         final MetacatJson metacatJson
     ) {
-        return new MysqlUserMetadataService(dataSourceManager, metacatJson, config);
+        return config.userMetadataServiceEnabled()
```

Comment:
Changed to profile based userMetadataService bean injection. This looks to be a very popular  approach to address this case, as we don't have much conditions to consider except picking up one implementation.